webgrid/__init__.py

```diff
@@ -133,7 +134,12 @@ def __init__(self, label, key=None, filter=None, can_sort=True,
         self.grid = None
         self.expr = None
         if render_in is not _None:
+            if callable(render_in):
```

Comment:
These callables are problematic here, due to the way webgrid instantiates columns (see `Column.new_instance`). When you set up your grid:
```
class MyGrid(Grid):
    Column('My Column', my_column_expr, ...)
```
Your column's `__init__` is run at declaration time, and you get a column instance. When the grid is used at runtime, we create runtime instances of these columns with some rather unfortunate magic, which copies attributes from the original "definition" instance to the new one. So, the callables here are being evaluated only once, the first time the grid module is imported, rather than each time the grid is used.

What we need to do is, pass through these attributes as-is. `render_in` will either be a tuple or a callable. Then I see two options:
- update all usage of `render_in` and `visible` to respect the callable. This seems inefficient.
- put a new method on the grid to resolve these lambdas, and call it in `new_instance` following the attribute assignments

Reply:
Resolved via a property, which allows assigning a lambda to the attribute at any time (not just in the constructor).